app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -162,6 +162,25 @@ def disk_mode(dependent, persistent)
     end
   end
 
+  def validate_device_backing(backing)
+    case backing.xsiType
+    when 'VirtualDiskFlatVer2BackingInfo'
```

Comment:
So they don't do a great job of documenting what modes are supported or not.
Those are all valid modes and the reconfigure will succeed, but if you actually use some of the modes the VM won't even power-on.